src/lib/formatter/RuleResultRecombinator.ts

```diff
@@ -11,50 +11,54 @@ import { constructSarif } from './SarifFormatter'
+	public static async recombineAndReformatResults(results: RuleResult[], outputOptions: OutputOptions, executedEngines: Set<string>): Promise<RecombinedRuleResults> {
 		// We need to change the results we were given into the desired final format.
 		let formattedResults: string | {columns; rows} = null;
-		switch (format) {
+		switch (outputOptions.format) {
 			case OUTPUT_FORMAT.CSV:
-				formattedResults = await this.constructCsv(results);
+				formattedResults = await this.constructCsv(results, outputOptions.normalizeSeverity);
 				break;
 			case OUTPUT_FORMAT.HTML:
-				formattedResults = await this.constructHtml(results);
+				formattedResults = await this.constructHtml(results, outputOptions.normalizeSeverity);
 				break;
 			case OUTPUT_FORMAT.JSON:
-				formattedResults = this.constructJson(results);
+				formattedResults = this.constructJson(results); // normalizeSeverity not needed because the json will display automatically if it sees the value
 				break;
 			case OUTPUT_FORMAT.JUNIT:
-				formattedResults = this.constructJunit(results);
+				formattedResults = this.constructJunit(results); // normalizeSeverity not needed because this output format doesn't include severity
 				break;
 			case OUTPUT_FORMAT.SARIF:
-				formattedResults = await constructSarif(results, executedEngines);
+				formattedResults = await constructSarif(results, executedEngines, outputOptions.normalizeSeverity);
 				break;
 			case OUTPUT_FORMAT.TABLE:
-				formattedResults = this.constructTable(results);
+				formattedResults = this.constructTable(results); // normalizeSeverity not needed because this output format doesn't include severity
 				break;
 			case OUTPUT_FORMAT.XML:
-				formattedResults = this.constructXml(results);
+				formattedResults = this.constructXml(results, outputOptions.normalizeSeverity);
 				break;
 			default:
 				throw new SfdxError('Unrecognized output format.');
 		}
-		return {minSev: this.findMinSev(results), results: formattedResults, summaryMap: this.generateSummaryMap(results, executedEngines)};
+		return {minSev: this.findMinSev(results, outputOptions.normalizeSeverity), results: formattedResults, summaryMap: this.generateSummaryMap(results, executedEngines)};
 	}
 
-	private static findMinSev(results: RuleResult[]): number {
+	private static findMinSev(results: RuleResult[], normalizeSeverity: boolean): number {
 		// If there are no results, then there are no errors.
 		if (!results || results.length === 0) {
 			return 0;
 		}
 		let minSev = null;
+		
+		// if -n or -s flag used, minSev is calculated with normal value
 		for (const res of results) {
 			for (const violation of res.violations) {
-				if (!minSev || violation.severity < minSev) {
-					minSev = violation.severity;
-				}
+			  var severity = (violation.normalizedSeverity == undefined)? violation.severity : violation.normalizedSeverity;
```

Comment:
In order to avoid accidental coercion, it's best to use `===` which forces strict comparison. This also applies to a few other places in this PR that are using `==`. In general it's best practice to use `===` unless for some reason coercion is needed.